reference_data/management/tests/update_omim_tests.py

```diff
@@ -0,0 +1,199 @@
+    def test_update_omim_command(self, mock_download, mock_logger):
+        mock_download.return_value = self.temp_file_path
+
+        # Test omim api response error
+        responses.add(responses.GET, 'https://api.omim.org/api/entry?apiKey=test_key&include=geneMap&format=json&mimNumber=612367',
+                      json={'error': 'not found'}, status=400)
+        # Test omim api responses with bad data
+        responses.add(responses.GET, 'https://api.omim.org/api/entry?apiKey=test_key&include=geneMap&format=json&mimNumber=612367',
+                      json={"omim": {"entryList": []}}, status=200)
+        # Normal omim api responses
+        responses.add(responses.GET, 'https://api.omim.org/api/entry?apiKey=test_key&include=geneMap&format=json&mimNumber=612367',
+                      json=OMIM_ENTRIES, status=200)
+
+        # Omim api response error test
+        with self.assertRaises(CommandError) as ce:
+            call_command('update_omim', '--omim-key=test_key')
+        self.assertEqual(ce.exception.message, 'Request failed with 400: Bad Request')
+
+        # Bad omim api response test
+        with self.assertRaises(CommandError) as ce:
+            call_command('update_omim', '--omim-key=test_key')
+        self.assertEqual(ce.exception.message, 'Expected 1 omim entries but recieved 0')
+
+        # Test without a file_path parameter
+        mock_logger.reset_mock()
+        call_command('update_omim', '--omim-key=test_key')
+
+        mock_download.assert_called_with('https://data.omim.org/downloads/test_key/genemap2.txt')
+
+        calls = [
+            mock.call('Deleting 0 existing Omim records'),
+            mock.call('Parsing file'),
+            mock.call('Creating 2 Omim records'),
+            mock.call('Done'),
+            mock.call('Loaded 2 Omim records from {}. Skipped 2 records with unrecognized genes.'.format(self.temp_file_path)),
+            mock.call('Running ./manage.py update_gencode to update the gencode version might fix missing genes')
+        ]
+        mock_logger.info.assert_has_calls(calls)
+
+        # test with a file_path parameter
+        mock_download.reset_mock()
+        mock_logger.reset_mock()
+        call_command('update_omim', '--omim-key=test_key', self.temp_file_path)
+        mock_download.assert_not_called()
+        calls = [
+            mock.call('Deleting 2 existing Omim records'),
+            mock.call('Parsing file'),
+            mock.call('Creating 2 Omim records'),
+            mock.call('Done'),
+            mock.call('Loaded 2 Omim records from {}. Skipped 2 records with unrecognized genes.'.format(self.temp_file_path)),
+            mock.call('Running ./manage.py update_gencode to update the gencode version might fix missing genes')
+        ]
+        mock_logger.info.assert_has_calls(calls)
+
+        self.assertEqual(Omim.objects.all().count(), 2)
+        record = Omim.objects.get(gene__gene_symbol = 'OR4F5')
+        self.assertDictEqual({
```

Comment:
This is pretty had to read, just do it as a line by line assertion
```
self.assertEqual(record.comments, 'linkage with rs1780324')
self.assertEqual(record. gene_description, Alkaline phosphatase, plasma level of, QTL 2')
...
```

Reply:
Okay